code/game/objects/items/flashlight.dm

```diff
@@ -69,7 +70,26 @@
 		to_chat(user, "<span class='notice'>You modify [src]. It can now be mounted on a weapon.</span>")
 		to_chat(user, "<span class='notice'>Use a screwdriver on [F] to change it back.</span>")
 		qdel(src) //Delete da old flashlight
+		
+/obj/item/flashlight/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	
+	if(istype(I, /obj/item/tool/wrench))
+		if(!underbarrel_light_compatible) //No fancy messages, just no
+			return
+
+		if(on)
+			to_chat(user, "<span class='warning'>Turn off [src] first.</span>")
+			return
+
+		if(loc == user)
+			user.dropItemToGround(src) //This part is important to make sure our light sources update, as it calls dropped()
 
+		var/obj/item/attachable/underbarrel_light/A = new(loc)
+		user.put_in_hands(A) //This proc tries right, left, then drops it all-in-one.
+		to_chat(user, "<span class='notice'>You modify [src]. It can now be mounted on a weapon.</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='notice'>You modify [src]. It can now be mounted on a weapon.<br>Use a wrench on [modified_light] to change it back.</span>")
```
You can call the proc a single time using `<br>` (line break) to add a new line.